### PR TITLE
ROX-11716: Initial work on step by step guide to setup OSD cluster

### PR DESCRIPTION
**Pull request by mtodor**
## Description

This PR introduces a new step-by-step guide to setting up an OSD cluster for development purposes.

## Checklist (Definition of Done)
- ~[ ] Unit and integration tests added~
- [x] Documentation added if necessary
- ~[ ] CI and all relevant tests are passing~

## Test manual

The best way to test is to execute all steps. Good luck! 😁

### CODE REVIEW

File: docs/setup-developer-osd-cluster.md [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+### Create OSD Cluster
+
+1. Create OSD Cluster with `ocm`
+
+Export name for your cluster. Prefix it with your initials or something similar to avoid name collisions. i.e. `mt-osd-1307`
+```
+export OSD_CLUSTER_NAME="<your cluster name>"
+```
+
+For staging OSD cluster, you should login to staging platform. You should use `rhacs-managed-service-dev` account.
```
Comment (kurlov) [inline docs/setup-developer-osd-cluster.md:29]: probably it's just me but `staging` is really confusing and maybe it is worth to mention that `staging` is a known value for ocm CLI

File: docs/setup-developer-osd-cluster.md [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+### Prepare cluster for RHACS Operator
+
+4. Export defaults
+```
+export RHACS_OPERATOR_CATALOG_VERSION="3.70.1"
+export RHACS_OPERATOR_CATALOG_NAME="redhat-operators"
+```
+
+4. Check if the latest version of available ACS Operator is high enough for you. If that is OK for you, you can skip next steps prefixed with `(ACS operator from branch)`.
```
Comment (kurlov) [inline docs/setup-developer-osd-cluster.md:95]: Numbering typo. It should be 5

**Review by kurlov — COMMENTED** [review]
Looks very well done 👍

File: docs/setup-developer-osd-cluster.md [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+grpcurl -plaintext -d '{"name":"rhacs-operator"}' localhost:50051 api.Registry/GetPackage | jq '.channels[0].csvName'
+```
+You can stop port-forward after this.
+
+6. (ACS operator from branch) Prepare pull secret
+**Important** This will change cluster wide pull secrets. It's not advised to use on clusters where credentials can be compromized.
+
+**Pay attention:** `docker-credential-osxkeychain` is specific for MacOS. For Linux please check `docker-credential-secretservice`.
+```
+export QUAY_REGISTRY_AUTH_BASIC=$(docker-credential-osxkeychain get <<<"https://quay.io" | jq -r '"\(.Username):\(.Secret)"')
```
Comment (mtesseract) [inline docs/setup-developer-osd-cluster.md:113]: Is this coming from Docker Desktop?

File: docs/setup-developer-osd-cluster.md [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+  namespace: openshift-marketplace
+spec:
+  displayName: 'RHACS Development'
+  publisher: 'Red Hat ACS'
+  sourceType: grpc
+  image: quay.io/rhacs-eng/stackrox-operator-index:v${RHACS_OPERATOR_CATALOG_VERSION}
+EOF
+```
+
+After what if you execute:
```
Comment (mtesseract) [inline docs/setup-developer-osd-cluster.md:145]:
```suggestion
By executing
```

**Review by kurlov — APPROVED** [review]

File: docs/setup-developer-osd-cluster.md (hunk 3 above)
Comment (mtodor) [reply docs/setup-developer-osd-cluster.md:113]: Probably it is. I have seen that in this doc on Stackrox repo: https://github.com/stackrox/stackrox/blob/master/operator/README.md#launch-the-operator-on-the-cluster-with-olm-and-the-bundle

I removed Docker Desktop a few days ago and switched to Rancher. Path to the command is `docker-credential-osxkeychain -> /Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/bin/docker-credential-osxkeychain`. So it's part of Rancher too.

We should not add secrets to the cluster anyway, we should switch to `stackrox-io` images. I don't expect this part to live long.

**Comment by mtesseract** [comment]
image
No idea...

**Review by mtesseract — APPROVED** [review]
I haven't executed the steps, but as I understand it others have, therefore: lgtm.

**Pull request merged**